src/espidf.rs

```diff
@@ -112,7 +112,7 @@ pub enum FromEnvError {
     #[error("`esp-idf` repository exists but required tools not in environment")]
     NotActivated {
         /// The esp-idf repository detected from the environment.
-        esp_idf_repo: git::Repository,
+        tree: SourceTree,
```

Comment:
I'd suggest instead of tree naming it something like `esp_idf_dir` (or leaving the old name).